File: dataset/human/cpp/file_1107.cpp
#include <iostream>
using namespace std;

int main()
{
  int r, g, b, n;
  cin >> r >> g >> b >> n;
  int ans(0);
  for (int i = 0; i <= n; i += r) {
    for (int j = 0; j <= n - i; j += g) {
      int k = n - i - j;
      if (k % b == 0) ans++;
    }
  }
  cout << ans << endl;
}

File: dataset/human/python/file_1107.py
import sys
from bisect import bisect_right, bisect_left
from collections import Counter


def main():
    input = sys.stdin.buffer.readline
    n = int(input())
    a = list(map(int, input().split()))
    b = list(map(int, input().split()))
    cnt = Counter(a + b)
    e, num = cnt.most_common(1)[0]
    if num > n:
        print("No")
    else:
        step = 0
        for e in set(a) & set(b):
            s = bisect_right(a, e) - bisect_left(b, e)
            step = max(step, s)
        ans = [0] * n
        for i in range(n):
            ans[i] = b[(i - step) % n]
        print("Yes")
        print(*ans)


if __name__ == "__main__":
    main()
